### PR TITLE
[MM-55122] Replaced the usage of LocalizedIcon in 'fa_warning_icon.tsx' with i/span tags

### LINKED ISSUE #25088: Replace usage of LocalizedIcon in 'fa_warning_icon.tsx' with i/span tags
Introduction

The purpose of this ticket is to eliminate the usage of LocalizedIcon in the `webapp/channels/src/components/widgets/icons/fa_warning_icon.tsx` file. LocalizedInput currently encourages the use of a custom 't' function, which deviates from the standard practices of the react-intl. We are transitioning to align all our translations with the principles and methods of react-intl, ensuring consistency throughout our codebase.

Before you start:

- Assignment Confirmation: Comment on this ticket to express your interest in owning it. This will help prevent task duplication.
- Environment Setup: Ensure your development environment is up and running. Follow the setup guide https://developers.mattermost.com/contribute/developer-setup
- Analyse: Navigate to the file referenced in this ticket and examine its usage within Mattermost. Capture screenshots of where and how LocalizedIcon(s) are used; these screenshots will be helpful for future comparisons in the PR.
- Best Practices: Familiarize yourself with the best practices for contributing to the Mattermost https://developers.mattermost.com/contribute/why-contribute

Guidelines During PR Development:

- Replace FormattedIcon Usage: If the LocalizedIcon has a prop-named `component` with the value `span` replace the component with a `<span/>` tag. Otherwise, use `<i/>` tag to replace the LocalizedIcon.
- Component Type: Determine whether the component you're modifying is a functional or a class component.
- For functional components, utilise the `useIntl` hook from react-intl to replace any instances of the `t` function with `formatMessage` for the component's title or aria label.
- For class components, apply the `injectIntl` Higher Order Component (HOC) from react-intl to use the `formatMessage` function. Replace any occurrences of the 't' function with `formatMessage` for the component's title or aria label.
- Testing: Visually verify that the changes didn't cause any alterations in the app's UI. Ensure any relevant tests are updated to reflect the changes made during this PR. In some cases, you might need to update snapshots. You can run `npm run test – -u` from the "channels" folder to update them.

Checklist Before PR Submission:

- Screenshots for Comparison: Capture screenshots of the places where you replaced LocalizedIcon. These screenshots will allow for a quick visual comparison within the PR, helping identify any unintended UI changes. Remember to include the screenshot you took earlier for a before-and-after comparison. Screenshots also make it easier for QA to locate and review the changes made to components.
- PR Title Format: Write the title of your PR using the following format: '[MM-jiraTicketNumber] Write the ticket title here'.
- CI Checks: Ensure all Continuous Integration checks have passed successfully. The review process can only commence when all CI checks are green.

For reference, you can follow this PR https://github.com/mattermost/mattermost/pull/25071

----
If you're interested please comment here and come [join our "Contributors" community channel](https://community.mattermost.com/core/channels/tickets) on our daily build server, where you can discuss questions with community members and the Mattermost core team. For technical advice or questions, please  [join our "Developers" community channel](https://community.mattermost.com/core/channels/developers).

New contributors please see our [Developer's Guide](https://developers.mattermost.com/contribute/getting-started/).

JIRA: https://mattermost.atlassian.net/browse/MM-55122

### PR DESCRIPTION
**Pull request by balajik**
#### Summary
Replaced the usage of LocalizedIcon in 'fa_warning_icon.tsx' with i tag

#### Ticket Link
Fixes #25088

Jira https://mattermost.atlassian.net/browse/MM-55122

#### Release Note

```release-note
NONE
```

### CODE REVIEW

**Comment by M-ZubairAhmed** [comment]
@balajik please follow the "PR checklist" guidelines mentioned in the ticket https://github.com/mattermost/mattermost/issues/25088#issue-1959024456

File: webapp/channels/src/components/widgets/icons/fa_warning_icon.tsx [hunk 1]
```diff
@@ -2,27 +2,28 @@
...
 type Props = {
-    additionalClassName: string | null;
+    additionalClassName?: string | null;
+    intl: IntlShape;
 }
 
-export default class WarningIcon extends React.PureComponent<Props> {
-    public static defaultProps: Props = {
+class WarningIcon extends React.PureComponent<Props> {
+    public static defaultProps: Partial<Props> = {
```
Comment (devinbinnie) [inline webapp/channels/src/components/widgets/icons/fa_warning_icon.tsx:14]: How come you added this `Partial` bit here?

**Review by devinbinnie — APPROVED** [review]
LGTM! Just one non-blocking question. Thanks @balajik!

File: webapp/channels/src/components/widgets/icons/fa_warning_icon.tsx (hunk 1 above)
Comment (devinbinnie) [reply webapp/channels/src/components/widgets/icons/fa_warning_icon.tsx:14]: You changed `Props` to `Partial<Props>`. Was there a reason why?

File: webapp/channels/src/components/widgets/icons/fa_warning_icon.tsx [hunk 2]
```diff
@@ -2,27 +2,28 @@
 // See LICENSE.txt for license information.
 
 import React from 'react';
-
-import LocalizedIcon from 'components/localized_icon';
-
-import {t} from 'utils/i18n';
+import {injectIntl} from 'react-intl';
+import type {IntlShape} from 'react-intl';
```
Comment (M-ZubairAhmed) [inline webapp/channels/src/components/widgets/icons/fa_warning_icon.tsx:6]:
```suggestion
import {injectIntl, type IntlShape} from 'react-intl';
```

File: webapp/channels/src/components/widgets/icons/fa_warning_icon.tsx [hunk 3]
```diff
@@ -2,27 +2,28 @@
...
 
 type Props = {
-    additionalClassName: string | null;
+    additionalClassName?: string | null;
```
Comment (M-ZubairAhmed) [inline webapp/channels/src/components/widgets/icons/fa_warning_icon.tsx:9]:
```suggestion
    additionalClassName?: string;
```

Comment (M-ZubairAhmed) [reply webapp/channels/src/components/widgets/icons/fa_warning_icon.tsx:9]: I dont think null is needed here,

**Review by M-ZubairAhmed — CHANGES_REQUESTED** [review]

File: webapp/channels/src/components/widgets/icons/fa_warning_icon.tsx [hunk 4]
```diff
@@ -2,27 +2,27 @@
...
 
-export default class WarningIcon extends React.PureComponent<Props> {
-    public static defaultProps: Props = {
+class WarningIcon extends React.PureComponent<Props> {
+    public static defaultProps: Partial<Props> = {
         additionalClassName: null,
     };
 
     public render(): JSX.Element {
         const className = 'fa fa-warning' + (this.props.additionalClassName ? ' ' + this.props.additionalClassName : '');
```
Comment (M-ZubairAhmed) [inline webapp/channels/src/components/widgets/icons/fa_warning_icon.tsx:18]: we can simplify here by using the `classname('fa fa-warning', this.props.additionalclass)`
the library will take care of the string concatenations and empty string checks. That way we don't have to add null as default value

**Review by M-ZubairAhmed — CHANGES_REQUESTED** [review]

File: webapp/channels/src/components/widgets/icons/fa_warning_icon.tsx (hunk 3 above)
Comment (M-ZubairAhmed) [reply webapp/channels/src/components/widgets/icons/fa_warning_icon.tsx]: please see my latest comment

**Review by M-ZubairAhmed — APPROVED** [review]
Thank you for going extra mile to improve the component as well as the ticket work 🤝

**Review by yasserfaraazkhan — APPROVED** [review]
image

**Pull request merged**